### PR TITLE
Asynchronous JS Scripts Loading 

### PR DESCRIPTION
**Pull request by ayaman**
This PR changes the way the message handler and the message console logger are loaded to options passed to "realtime_support". To load the message handler ```realtime_support(message_handler: true)``` should be used, and for the message console logger, the ```message_console_logger: true``` option should be used.

It also adds the option to load external JS scripts asynchronously using the ```async: true``` option. This allows others scripts and elements on the page to load without waiting for the realtime scripts. To know when realtime functionality is available, the user should listen for the ```realtime:loaded``` event on the Document.

In terms of browser support, all functions should be cross-browser but unfortunately I don't have IE to make sure it's all working properly.

### CODE REVIEW

**Comment by mikeatlas** [comment]
:+1: looks good from a quick glance, I hope to test it out quickly this weekend and merge. Nice work and thank you so much for the contribution!

**Comment by ayaman** [comment]
Sounds good! Are you planning to release a new gem version with the updates once they're merged?

**Comment by mikeatlas** [comment]
Most likely, yes, and will add your name to the contributors if you like. I've been meaning to check if the changes you made are semver minor version compatible (upgrading does not affect existing functionality) or if it needs a major version. 

Truthfully I've transitioned out of Rails world in a new role in the last week and don't have an environment ready to try out the changes you contributed quickly - that's the only thing slowing me down from accepting the PR right away.

File: lib/realtime/view_helpers.rb [hunk 1]
```diff
@@ -1,26 +1,17 @@
...
+      return render(template: "realtime/#{async}realtime_support",
                     layout: nil,
                     locals:
                     { realtime_token: @realtime_token,
                       realtime_domain: @realtime_domain,
                       realtime_server_url: @realtime_server_url,
-                      realtime_user_id: @realtime_user_id }).to_s
-    end
-
-    def realtime_message_handler
```
Comment (mikeatlas) [inline lib/realtime/view_helpers.rb]: @ayaman I believe removing these view helpers (`realtime_message_handler ` and `realtime_message_console_logger `) and turning them into optional arguments makes the PR backwards-incompatible, requiring a major-version bump (I guess 1.0? With semver it's presently at 0 for the major version, 0.0.12). 

That's not to say your changes aren't a good idea, but anyone who might actually be depending on these still existing during a blindly-trusting `bundle update` run should expect the gem not to break their app.

Maybe you can put these back in, but also before the `return`, you add a `ActiveSupport::Deprecation.warn("deprecation warning")`?

**Comment by ayaman** [comment]
I added the methods back with a deprecation warning as you suggested. This should maintain backward compatibility and allow the methods to be easily removed later.

**Comment by mikeatlas** [comment]
Thanks for the patience @ayaman, I finally merged and pushed a new minor version of the gem, and also added you as a collaborator and appreciate the work you put into #8! Since there are no tests in the gem (it's mostly an "integration" project and tests would have mock out a lot of things probably) I presumed that bumping it up to minor version 0.1.0 was correct. Also pushed to RubyGems.

**Comment by mikeatlas** [comment]
(and if you want to be a gemcutter on rubygems, let me know which email address you use there, or register an account on rubygems.org and let me know if it's not the same email as the one on your profile here on github)

**Comment by ayaman** [comment]
Thank you @mikeatlas and no worries about the delay at all :-) My account on rubygems uses the same email as that on github actually.

**Comment by mikeatlas** [comment]
@ayaman You can cut new versions of the gem now if you like or as needed along with full collaborator on the repo. I personally am not working in RoR anymore and if people are still finding this project useful, having additional stakeholders is really helpful in keeping it going forward as needed - otherwise I personally wouldn't be spending more time maintaining it at this point. If you're not familiar with the rules of Semantic Versioning, (semver) have a read up... http://semver.org/ - minor version numbers are considered non-breaking upgrades, major versions can break or change functionality completely. Build numbers presume complete compatibility.

```
gem owner realtime -a [ayaman email]
Pushing gem to https://rubygems.org...
Owner added successfully.
Pushing gem to https://rubygems.org...
Owners for gem: realtime
- [ayaman email]
```

**Pull request merged**